src/h3lib/include/mathExtensions.h

```diff
@@ -20,12 +20,14 @@
 #ifndef MATHEXTENSIONS_H
 #define MATHEXTENSIONS_H
 
+#include <stdint.h>
+
 /**
  * MAX returns the maximum of two values.
  */
 #define MAX(a, b) (((a) > (b)) ? (a) : (b))
 
 // Internal functions
-int _ipow(int base, int exp);
+uint64_t _ipow(uint64_t base, uint64_t exp);
```

Comment:
Total nit, but why do you need a 64-bit exponent?

Reply:
Agreed that `uint64_t` seems like a larger size than what'd you'd ever need (and most of its larger values would produce incorrect results), but I wasn't sure what smaller size to choose, so I defaulted to the output size.

I do like that it describes the constraint that the exponent should be nonnegative (negative values give you fractional results instead of integral). However, maybe that isn't a good use of the type system, and we should just add a non-negativity check for the input...

Similarly, I don't think we have a use case for the base to be a negative integer, so I also chose that to be unsigned. But again, maybe that's an abuse of the type system. Definitely interested to hear other folks thoughts here.

I'm also not sure if `uint64_t` is the best output type. Subtraction with another `uint64_t` will remain a `uint64_t` and thus wrap around. Since we may do arithmetic with resolutions and counts of cells, maybe these outputs should be `int64_t` to avoid arithmetical surprises?

Reply:
Also, for the `exp` parameter, I think it needs to be `unsigned` of some sort, as the bit-shifting logic in the `_ipow` function won't work correctly for negative numbers.

Reply:
I've decided to go with the signature `int64_t _ipow(uint64_t base, uint64_t exp)`. Let me know if you all have any thoughts!